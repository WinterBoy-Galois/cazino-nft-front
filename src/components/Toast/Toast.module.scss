@import '../../styles/shared';

.container {
  display: flex;
  align-items: center;
}

.icon {
  margin-right: 1rem;
}
