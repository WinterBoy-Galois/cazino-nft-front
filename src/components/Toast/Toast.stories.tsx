import React from 'react';
import { text, button, boolean } from '@storybook/addon-knobs';
import { ToastContainer, info, error } from '.';
import { success } from '.';

export default {
  title: 'Components/Toast',
  component: ToastContainer,
};

const data = {
  message: 'Hello World!',
};

export const Default = () => {
  text('Toast text', data.message);
  button('Show success toast', () => success(text('Toast text', data.message)));
  button('Show info toast', () => info(text('Toast text', data.message)));
  button('Show error toast', () => error(text('Toast text', data.message)));

  return (
    <div>
      <ToastContainer autoClose={boolean('Auto close', true)} />
    </div>
  );
};
