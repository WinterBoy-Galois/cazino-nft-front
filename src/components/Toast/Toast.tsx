import React, { useEffect } from 'react';
import styles from './Toast.module.scss';
import Info from '../icons/Info';
import Check from '../icons/Check';
import Error from '../icons/Error';

interface IProps {
  type?: 'SUCCESS' | 'INFO' | 'ERROR';
}

const Toast: React.FC<IProps> = ({ children, type = 'INFO' }) => {
  let icon;

  switch (type) {
    case 'INFO':
      icon = <Info />;
      break;

    case 'SUCCESS':
      icon = <Check />;
      break;

    case 'ERROR':
      icon = <Error />;
      break;
  }

  return (
    <div className={styles.container}>
      <div className={styles.icon}>{icon}</div>
      <div>{children}</div>
    </div>
  );
};

export default Toast;
