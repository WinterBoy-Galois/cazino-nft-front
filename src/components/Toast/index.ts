export { default as ToastContainer } from './components/ToastContainer';
export * from './lib/toast';
