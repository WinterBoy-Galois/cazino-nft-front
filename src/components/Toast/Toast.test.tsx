import React from 'react';
import { render } from '@testing-library/react';
import Toast from './Toast';

describe('Toast', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<Toast />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
