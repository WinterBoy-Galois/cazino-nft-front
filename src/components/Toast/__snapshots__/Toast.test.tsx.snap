// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="icon"
        >
          <svg
            class="container"
            viewBox="0 0 31.02 31.02"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <g>
                <path
                  class="fill"
                  d="M15.51 0A15.51 15.51 0 1031 15.51 15.5 15.5 0 0015.51 0zm0 6.88a2.63 2.63 0 11-2.63 2.63 2.63 2.63 0 012.63-2.63zM19 22.76a.75.75 0 01-.75.75h-5.5a.75.75 0 01-.75-.75v-1.5a.76.76 0 01.75-.75h.75v-4h-.75a.75.75 0 01-.75-.75v-1.5a.76.76 0 01.75-.75h4a.76.76 0 01.75.75v6.26h.75a.75.75 0 01.75.75z"
                />
              </g>
            </g>
          </svg>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="icon"
      >
        <svg
          class="container"
          viewBox="0 0 31.02 31.02"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <g>
              <path
                class="fill"
                d="M15.51 0A15.51 15.51 0 1031 15.51 15.5 15.5 0 0015.51 0zm0 6.88a2.63 2.63 0 11-2.63 2.63 2.63 2.63 0 012.63-2.63zM19 22.76a.75.75 0 01-.75.75h-5.5a.75.75 0 01-.75-.75v-1.5a.76.76 0 01.75-.75h.75v-4h-.75a.75.75 0 01-.75-.75v-1.5a.76.76 0 01.75-.75h4a.76.76 0 01.75.75v6.26h.75a.75.75 0 01.75.75z"
              />
            </g>
          </g>
        </svg>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
